### ISSUE TITLE
[Feature Request] Livewire magic actions

### ISSUE THREAD

**Issue by DanielSpravtsev** [laravel-idea/plugin#218]
Blade wire:click magic actions:
$refresh : void
$set(property, value)
$toggle(component public property) : bool 
$emit(component function name, params)
[Docs](https://laravel-livewire.com/docs/2.x/actions)

Blade example:
```blade
        <x-section.heading title="Wallet">
            <x-button.primary wire:click="$toggle('walletModal')">Create</x-button.primary>
        </x-section.heading>
```

Magic actions can be also used in component listeners
Example:
```php
class LivewireComponentName extends Component
{
   protected $listeners = [
        'event-name' => '$refresh',
        'event-name-2' => 'component function name'
    ]; 

    or

    public function getListeners(){
        return [
            'event-name' => '$refresh'
        ];
    }
}
```

**Comment by adelf** [comment]
It's mighty magic :) it needs Javascript language injecting and some other stuff. I will probably collect some livewire feature requests and implement them together.

**Comment by DanielSpravtsev** [comment]
> It's mighty magic :) it needs Javascript language injecting and some other stuff. I will probably collect some livewire feature requests and implement them together.

I don't think so, you can just collect all livewire magic and then suggest them on specific actions. Anyway, if you want to go deeper, then I am all for it 😄

**Comment by adelf** [comment]
Well, at least the first part is implemented in the 8.2.2. What about listeners? Are they widely used?
